src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -703,6 +704,25 @@ public DatasetSummary mapRow(ResultSet rs, int rowNum) throws SQLException {
     }
   }
 
+  /**
+   * Update a dataset according to specified fields in the patch request. Any fields unspecified in
+   * the request will remain unaltered.
+   *
+   * @param id dataset UUID
+   * @param patchRequest updates to merge with existing dataset
+   * @return whether the dataset record was updated
+   */
+  @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+  public boolean patch(UUID id, DatasetPatchRequestModel patchRequest) {
```

Comment:
One thing that I haven't added yet is any kind of logging, debug or otherwise. I welcome suggestions or examples on what might be helpful (and can also talk to Ruchi about audit trail requirements that may come down the pike).

Reply:
I think an audit log would be helpful to indicate the updates made (Just a logger.info() statement). Although it might be tricky to report which fields are actually updated since it relies on COALESCE to determine whether the field is null in the patch request.

Reply:
I added in logging on successful patches, I kept the logging statement simple in that I'm logging the stringified patch request.  While it doesn't exclude any null fields, I think it meets our needs for auditing for now.  Here's an example log from a successful snapshot patch:

```
{"timestampSeconds":1653318344,"timestampNanos":124000000,"severity":"INFO","message":"Snapshot 6510b127-c226-46a1-ac12-c5f2383cbd08 patched with class SnapshotPatchRequestModel {\n    consentCode: c03\n}","serviceContext":{"service":null,"version":null},"context":"default","thread":"http-nio-8080-exec-5","logger":"bio.terra.service.snapshot.SnapshotDao","logging.googleapis.com/sourceLocation":{"file":"bio/terra/service/snapshot/SnapshotDao.java","line":701,"function":"bio.terra.service.snapshot.SnapshotDao.patch"},"logging.googleapis.com/trace":"projects/broad-jade-dev/traces/8f0a46e8b13f1756181b044de2d92210","logging.googleapis.com/spanId":"38c45d15a2f2c01e","logging.googleapis.com/trace_sampled":false,"traceId":"616f81b02ad57bba","spanId":"616f81b02ad57bba","requestId":"6WYvDwyW"}
```